include/nbl/video/utilities/IUtilities.h

```diff
@@ -26,6 +26,29 @@ class NBL_API IUtilities : public core::IReferenceCounted
         {
             auto physicalDevice = m_device->getPhysicalDevice();
             const auto& limits = physicalDevice->getLimits();
+
+            auto queueFamProps = physicalDevice->getQueueFamilyProperties();
+            uint32_t minImageTransferGranularityVolume = 1u; // minImageTransferGranularity.width * height * depth
+
+            for (uint32_t i = 0; i < queueFamProps.size(); i++)
+            {
+                uint32_t volume = queueFamProps[i].minImageTransferGranularity.width * queueFamProps[i].minImageTransferGranularity.height * queueFamProps[i].minImageTransferGranularity.depth;
+                if(minImageTransferGranularityVolume < volume)
+                    minImageTransferGranularityVolume = volume;
+            }
+
+            const uint32_t bufferOptimalTransferAtom = limits.maxResidentInvocations*sizeof(uint32_t);
+            const uint32_t imageOptimalTransferAtom = limits.maxResidentInvocations * asset::TexelBlockInfo(asset::EF_R64G64B64A64_SFLOAT).getBlockByteSize() * minImageTransferGranularityVolume;
+            const uint32_t optimalTransferAtom = core::max(bufferOptimalTransferAtom, imageOptimalTransferAtom);
+
+            // nonCoherentAtomSize < minBlockSize < optimalTransferAtom < stagingBuffeRSize/4
+            assert(limits.nonCoherentAtomSize < minStreamingBufferAllocationSize);
+            assert(minStreamingBufferAllocationSize < optimalTransferAtom);
+            assert(optimalTransferAtom * 4u < upstreamSize);
+            assert(optimalTransferAtom * 4u < downstreamSize);
+            assert(static_cast<uint64_t>(minStreamingBufferAllocationSize) % limits.nonCoherentAtomSize == 0u);
```

Comment:
if you do https://github.com/Devsh-Graphics-Programming/Nabla/pull/389/files#r954978095

then you need to substitute nonCoherentAtomSize for the max of itself and the optimal buffer transfer alignment